.github/workflows/ci.yml

```diff
@@ -92,31 +93,57 @@ jobs:
         id: version
         run: .github/workflows/docker-version.sh
 
+      - name: docker-meta-webserver
+        id: meta-webserver
+        uses: docker/metadata-action@v3
+        with:
+          images: |
+            sourcegraph/zoekt-webserver
+          tags: |
+            type=ref,event=branch
+            type=ref,event=pr
+            type=semver,pattern={{version}}
+            type=sha
+      - name: docker-meta-indexserver
+        id: meta-indexserver
+        uses: docker/metadata-action@v3
+        with:
+          images: |
+            sourcegraph/zoekt-indexserver
+          tags: |
+            type=ref,event=branch
+            type=ref,event=pr
+            type=semver,pattern={{version}}
+            type=sha
+
       - name: build-zoekt
-        uses: docker/build-push-action@v1
+        uses: docker/build-push-action@v4
         with:
-          repository: zoekt
-          tags: "latest"
-          add_git_labels: "true"
+          context: .
+          tags: "zoekt:latest"
           push: "false"
-          build_args: VERSION=${{ steps.version.outputs.value }}
+          build-args: VERSION=${{ steps.version.outputs.value }}
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
 
       - name: build-push-webserver
-        uses: docker/build-push-action@v1
+        uses: docker/build-push-action@v4
         with:
-          repository: sourcegraph/zoekt-webserver
-          tags: ${{ steps.version.outputs.value }},latest
-          dockerfile: Dockerfile.webserver
-          add_git_labels: "true"
-          username: ${{ secrets.DOCKER_USERNAME }}
-          password: ${{ secrets.DOCKER_PASSWORD }}
+          context: .
+          tags: sourcegraph/zoekt-webserver:${{ steps.version.outputs.value }}, ${{ steps.meta-webserver.outputs.tags }}, sourcegraph/zoekt-webserver:latest
+          file: Dockerfile.webserver
+          cache-from: sourcegraph/zoekt-webserver:latest
+          push : "${{ github.branch == 'main' }}"
```

Comment:
does this still use the image build in step `build-zoekt`? The cache-from makes me slightly concerns this is working accidently and isn't actually using the artifacts form this commit.

Alternatively it is just rebuilding the build-zoekt image twice between webserver and indexserver? Just trying to understand. Do you have an example images I can pull to confirm?

Reply:
Yes, it does, I initially forgot to tag properly the image in `build-zoekt` and it complained of not finding out `library/zoekt` which is the tag for locally build images.